### ISSUE TITLE
Fatal signal 6 (SIGABRT) crash on android release when forcing screen orientation

### ISSUE THREAD

**Issue by H-A-Hristov** [dotnet/maui#22106]
### Description

Fatal signal 6 (SIGABRT) crash in logcat, when running release build on android 12 device, using forced orientation and activated concurrent GC.

Debug version works.
Not specifying orientation works.
Disabling concurrent GC works.

### Steps to Reproduce

1. Android release build.
2. Concurrent GC turned on.
3. ScreenOrientation = ScreenOrientation.Portrait set as annotation to the Main activity.

### Version with bug

8.0.7/8.0.100

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 12

### Did you find any workaround?

Debug version works.
Not specifying orientation works.
Disabling concurrent GC works.

**Comment by PureWeen** [comment]
Can you attach a logcat file with your crash?
https://learn.microsoft.com/en-us/xamarin/android/deploy-test/debugging/android-debug-log?tabs=windows

**Comment by MAUIoxo** [comment]
I have a similar issue when changing Orientation, but in combination with the Community Toolkit Popup:

![IMG_0426](https://github.com/dotnet/maui/assets/140683727/f2776053-64ce-4dcc-bb03-870f1436c6f7)

This results in the following Exception:

![IMG_0427](https://github.com/dotnet/maui/assets/140683727/30c7d215-a904-4bbc-8758-f3c15a1a9345)

This came since a couple of Nightly builds and used to work previously, but strangely enough also in combination with changed Orientation. I don’t change it in Code-Behind, but normally in Simulator or on a real Device, both crash.

Was there a change having an impact on changed orientation, lately?

**Comment by PureWeen** [comment]
@MAUIoxo that might be different
Can you log that with MCT?

**Comment by MAUIoxo** [comment]
I can't, but in Debug Mode, at least for my issue, I got the following Stacktrace info:

```cs
   at CommunityToolkit.Maui.Core.Views.MauiPopup.SetShadowView(UIView& target)
   at CommunityToolkit.Maui.Core.Views.MauiPopup.ViewDidLayoutSubviews()
   at UIKit.UIApplication.UIApplicationMain(Int32 , String[] , IntPtr , IntPtr ) in /Users/builder/azdo/_work/1/s/xamarin-macios/src/UIKit/UIApplication.cs:line 58
   at UIKit.UIApplication.Main(String[] , Type , Type ) in /Users/builder/azdo/_work/1/s/xamarin-macios/src/UIKit/UIApplication.cs:line 94
   at MacroOptimizerApp.Program.Main(String[] args) in D:\Development\C#\OptimizerApp\OptimizerApp\Platforms\iOS\Program.cs:line 13
```

**Comment by H-A-Hristov** [comment]
Update on the matter:
I got a suggestion to disable trimming (and AOT compilations respectively) and test again.
And so I did, combined with re-enabling the concurrent GC. 

This solved the issue.
The more interesting part - even enabling trimming and AOT, I cannot reproduce the problem again.
I was able to reproduce it at will, by adding either screen orientation, or the concurrent GC option.

[log.txt](https://github.com/dotnet/maui/files/15147788/log.txt)

Anyway, I plan to accept my miracle and move on.
